jwst/regtest/test_nircam_image.py

```diff
@@ -0,0 +1,115 @@
+import os
+
+import pytest
+from astropy.io.fits.diff import FITSDiff
+from astropy.table import Table, setdiff
+
+from jwst.pipeline.collect_pipeline_cfgs import collect_pipeline_cfgs
+from jwst.stpipe import Step
+
+
+@pytest.fixture(scope="module")
+def run_pipelines(jail, rtdata_module):
+    """Run stages 1-3 pipelines on NIRCam imaging data."""
+    rtdata = rtdata_module
+    rtdata.get_data("nircam/image/jw42424001001_01101_00001_nrca5_uncal.fits")
+
+    collect_pipeline_cfgs("config")
+
+    # Run detector1 pipeline only on one of the _uncal files
+    args = ["config/calwebb_detector1.cfg", rtdata.input,
+        "--steps.dq_init.save_results=True",
+        "--steps.saturation.save_results=True",
+        "--steps.superbias.save_results=True",
+        "--steps.refpix.save_results=True",
+        "--steps.linearity.save_results=True",
+        "--steps.dark_current.save_results=True",
+        "--steps.jump.save_results=True",
+        "--steps.jump.rejection_threshold=50.0",
+        ]
+    Step.from_cmdline(args)
+
+    # And run image2 pipeline on the produced _rate file
+    rtdata.input = "jw42424001001_01101_00001_nrca5_rate.fits"
+    args = ["config/calwebb_image2.cfg", rtdata.input,
+        "--steps.assign_wcs.save_results=True",
+        ]
+    Step.from_cmdline(args)
+
+    # Grab rest of _rate files for the asn and run image2 pipeline on each to
+    # produce fresh _cal files for the image3 pipeline.  We won't check these
+    # or look at intermediate products
+    rate_files = [
+    "nircam/image/jw42424001001_01101_00001_nrcb5_rate.fits",
+    "nircam/image/jw42424001001_01101_00002_nrca5_rate.fits",
+    "nircam/image/jw42424001001_01101_00002_nrcb5_rate.fits",
+    "nircam/image/jw42424001001_01101_00003_nrca5_rate.fits",
+    "nircam/image/jw42424001001_01101_00003_nrcb5_rate.fits",
+    ]
+    for rate_file in rate_files:
+        rtdata.get_data(rate_file)
+        args = ["config/calwebb_image2.cfg", rtdata.input,
+            "--steps.resample.skip=True"]
+        Step.from_cmdline(args)
+
+    # Get the level3 assocation json file (though not its members) and run
+    # image3 pipeline on all _cal files listed in association
+    rtdata.get_data("nircam/image/jw42424-o002_20191220t214154_image3_001_asn.json")
+    args = ["config/calwebb_image3.cfg", rtdata.input,
+        # Comment out following lines, as the dataset is currently broken
+        # "--steps.tweakreg.save_results=True",
+        # "--steps.skymatch.save_results=True",
+        ]
+    Step.from_cmdline(args)
+
+    return rtdata
+
+
+@pytest.mark.bigdata
+@pytest.mark.parametrize("suffix", ["dq_init", "saturation", "superbias",
+    "refpix", "linearity", "trapsfilled", "dark_current", "jump", "rate",
```

Comment:
If the data have NINTS>1 you could also include the _rateints file in the comparison. Even though it isn't used downstream, it's still useful to test as another output of the ``ramp_fit`` step.

Reply:
NINTS=1 for this dataset.